src/mudlet.cpp

```diff
@@ -2433,22 +2433,34 @@ void mudlet::playSound( QString s, int soundVolume )
     if (!pPlayer) {
         pPlayer = new QMediaPlayer(this);
 
+        auto pHost = getActiveHost();
+
         if (!pPlayer) {
             /* It (should) be impossible to ever reach this */
-            Host * pH = getActiveHost();
-            
-            if (pH) {
-                pH->postMessage( "\n[  ERROR  ]  - Unable to create new QMediaPlayer object\n" );    
+            if (pHost) {
+                pHost->postMessage("\n[  ERROR  ]  - Unable to create new QMediaPlayer object\n");
             }
-            
             return;
         }
 
-        MusicBoxList.append(pPlayer);
+        connect(pPlayer, &QMediaPlayer::stateChanged, [=](QMediaPlayer::State state) {
```

Comment:
Definitely not. Creating them is **expensive** - taking on the order of 30 - 100ms for each object and we re-use them anyway, so it's not like playing another sound is going to create another QMediaPlayer.